src/shared/components/help-popup/HelpPopupButton.tsx

```diff
@@ -34,6 +34,15 @@ const HelpPopupButton = (props: Props) => {
   const [shouldShowModal, setShouldShowModal] = useState(false);
 
   const openModal = () => {
+    const trackHelpPopupOpen = new CustomEvent('analytics', {
+      detail: {
+        category: 'contextual-help',
+        action: 'opened'
+      }
+    });
+
+    window.dispatchEvent(trackHelpPopupOpen);
```

Comment:
Here are a couple of things I don't like about this:

- it's dispatched against the window rather than against the DOM node of this element
- it makes HelpPopupButton aware of:
  - analytics in general (yuck!)
  - Google analytics in particular (because of the payload) — double yuck!

Imagine HelpPopupButton were a component of a design system separate from the `ensembl-client` codebase.

We could, potentially, produce an event called something like `ensembl-component-info` (or something better, but sufficiently non-analyticsy), with a payload that looks something like:

```
{
  componentName: 'help-popup-button', // or 'help-popup'
  action: 'help-shown',
  resourceType: 'article',
  resourceName: '<title of the article or of the video>'
}
```
The downside, obviously, is that there will need to be code that transforms this into Google analytics events. The upside is that this code won't be in the HelpPopupButton.

Reply:
Why do we need to hide the actual purpose of events like this from these components?

Analytics is the only purpose these events are dispatched.

When someone reads this code in future, this event being `non-analyticsy` will only increase the time to understand what is going on here.

Reply:
> When someone reads this code in future, this event being non-analyticsy will only increase the time to understand what is going on here.

On the other hand, if we extracted these components into a components library, they would have these weird analytics events associated with them and tailored to a very specific purpose — reporting to Google analytics.

I suppose one could look at such components as being more chatty than others in terms of what they want to tell the world about themselves.

Reply:
Even if these components are moved to a components library, are these events going to be used for something other than analytics?

I think these events should be dispatched only if they are used for something.